### DISCUSSION TITLE
remove-button inside blueprintjs table cell

### DISCUSSION THREAD

**Discussion by bsdis** [palantir/blueprint#5875] [Q&A] [answered]
I need a way for the user to remove rows from a blueprintjs table. The best solution i could think of for this interaction is to add a column with a button in the cell
```
<Cell>
          <Button icon="remove" />
 </Cell>
```
However this seems to not work for the following reasons
<img width="330" alt="Screenshot 2020-07-18 at 14 00 18" src="https://user-images.githubusercontent.com/1256202/87852338-5a93b000-c901-11ea-856d-edd78f2c486a.png">

Cell cuts the buttons upper and lower part instead of increasing the height of the cell. Also when the button is clicked, the cell is also selected which is problematic. Further the button is not centered and the column is not packing around the button, which would look better if it did. See above screenshot. 
How is this problem solved?

**Comment by adidahiya** [comment, marked as answer]
Buttons aren't designed to fit inside table cells, as the cells should represent data. If you want to implement this kind of "remove" behavior I would recommend rendering an `<Icon>` inside the cell. Then try adding custom styling and click handling behavior (using `onSelection`) for the "remove" column cells.
